pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -2703,7 +2704,12 @@ var (
 		}
 	}
 	addPodActiveQ = func(t *testing.T, logger klog.Logger, queue *PriorityQueue, pInfo *framework.QueuedPodInfo) {
-		queue.activeQ.unlocked().AddOrUpdate(pInfo)
+		if !pInfo.Timestamp.IsZero() {
+			timeBefore := queue.clock.Now()
+			defer queue.clock.(*testingclock.FakeClock).SetTime(timeBefore)
+			queue.clock.(*testingclock.FakeClock).SetTime(pInfo.Timestamp)
+		}
+		queue.Add(logger, pInfo.Pod)
```

Comment:
In this case, I prefer we use `underLock`, with a comment that we are influencing the time when it was added.

Reply:
Done